napari/utils/transforms/transforms.py

```diff
@@ -597,6 +628,14 @@ def expand_dims(self, axes: Sequence[int]) -> 'Affine':
             name=self.name,
         )
 
+    @cached_property
+    def is_diagonal(self):
+        return is_diagonal(self.linear_matrix, tol=1e-8)
```

Comment:
Would be good to document that the diagonal check does not include the translation component of the whole affine matrix.